include/zephyr/net/wifi.h

```diff
@@ -536,6 +536,40 @@ static inline const char *get_ps_config_err_code_str(int16_t err_no)
 	return "<unknown>";
 }
 
+/** Promiscuous mode operational settings */
+enum wifi_promiscuous_mode {
+	/** Disable promiscuous Mode setting */
+	WIFI_PROMISC_DISABLE = 0,
+	/** Connection less sniffer mode setting enable */
+	WIFI_PROMISC_CONN_LESS_MODE,
+	/** STA sniffer mode setting enable */
+	WIFI_PROMISC_STA_MODE,
+	/** AP sniffer mode setting enable */
+	WIFI_PROMISC_AP_MODE,
+	/** STA-AP sniffer mode setting enable */
+	WIFI_PROMISC_STA_AP_MODE,
+};
+
+/** Promiscuous mode filter settings */
+enum wifi_promiscuous_filter {
+	/** Support mgmt/data and ctrl packet sniffing */
+	WIFI_PROMISC_ALL = 1,
+	/** Support only sniffing of mgmt packets */
+	WIFI_PROMISC_MGMT = 2,
+	/** Support only sniffing of data packets */
+	WIFI_PROMISC_DATA = 4,
+	/** Support only sniffing od ctrl packets */
+	WIFI_PROMISC_CTRL = 8,
+};
+
+/** Promiscuous mode/filter command operation */
+enum wifi_promisc_op {
```

Comment:
Nit: Try to be consistent in naming, the other enums use `promiscuous`.

Reply:
will address the change